lib/auth/clt.go

```diff
@@ -352,6 +352,11 @@ func (c *Client) CreateRemoteCluster(ctx context.Context, rc types.RemoteCluster
 	return nil, trace.NotImplemented(notImplementedMessage)
 }
 
+// PatchRemoteCluster not implemented: can only be called locally.
```

Comment:
Why?

Reply:
So there's two prongs for how we ended up here.

Firstly, the Patch method is something that'll only be implemented on the Backend store and not available over gRPC. To get the Patch behaviour over gRPC, instead the Update RPC is called and an Update Mask is provided to specify which fields to change. This is mostly down to the fact that the Patch method's signature is not something that's easily converted to gRPC (updateFn).

The "why" of why we have to add a pointless stub method to `Client` is a bit more historical. We have a bunch of heavily shared interfaces for the Backend stores that all embed into the `auth.ClientI` interface, which is then heavily shared over the codebase. This comes from a time where there was a concerted effort to make all the layers share the same signature and methods (e.g the HTTP client methods match the methods you'd use internally to access the store) - this has ended up causing a fair bit of pain since you don't always want the behaviour of your external API to match the behaviour of your internal methods, so we're starting to move away from this now. Unfortunately, this does mean we're in a situation at the moment where in order to make a certain method callable within the Auth Server internally, it has to be added as a method to the Client - hence these ugly stubs. We're making fairly ok, but slow, progress towards decoupling these interfaces and the switch to pure gRPC is part of this.